src/main/java/com/questhelper/skills/mining/Mining.java

```diff
@@ -0,0 +1,191 @@
+		fullTraining.addStep(new Conditions(inCopperZone), mineCopperOrTin);
+		fullTraining.addStep(new Conditions(mi15, inIronZone), mineIron);
+		fullTraining.addStep(mi15, ironStep);
+		return fullTraining;
+	}
+
+	@Override
+	public void setupRequirements()
+	{
+		mi6 = new SkillRequirement(Skill.MINING, 6);
+		mi11 = new SkillRequirement(Skill.MINING, 11);
+		mi21 = new SkillRequirement(Skill.MINING, 21);
+		mi31 = new SkillRequirement(Skill.MINING, 31);
+		mi41 = new SkillRequirement(Skill.MINING, 41);
+
+		mi15 = new SkillRequirement(Skill.MINING, 15);
+
+
+		ironPickaxe = new ItemRequirement("Iron pickaxe", ItemID.IRON_PICKAXE);
+		steelPickaxe = new ItemRequirement("Steel pickaxe", ItemID.STEEL_PICKAXE);
+		blackPickaxe = new ItemRequirement("Black pickaxe", ItemID.BLACK_PICKAXE);
+		mithrilPickaxe = new ItemRequirement("Mithril pickaxe", ItemID.MITHRIL_PICKAXE);
+		adamantPickaxe = new ItemRequirement("Adamant pickaxe", ItemID.ADAMANT_PICKAXE);
+		runePickaxe = new ItemRequirement("Rune pickaxe", ItemID.RUNE_PICKAXE);
+
+		ironPickaxe = ironPickaxe.showConditioned(new Conditions(LogicType.NOR, at5, mi6)
+		).isNotConsumed();
+		steelPickaxe = steelPickaxe.showConditioned(
+			new Conditions(at5, mi6, new Conditions(LogicType.NOR, at10, mi11))
+		).isNotConsumed();
+		blackPickaxe = blackPickaxe.showConditioned(
+			new Conditions(at10, mi11, new Conditions(LogicType.NOR, at20, mi21))
+		).isNotConsumed();
+		mithrilPickaxe = mithrilPickaxe.showConditioned(
+			new Conditions(at20, mi21, new Conditions(LogicType.NOR, at30, mi31))
+		).isNotConsumed();
+		adamantPickaxe = adamantPickaxe.showConditioned(
+			new Conditions(at30, mi31, new Conditions(LogicType.NOR, at40, mi41))
+		).isNotConsumed();
+		runePickaxe = runePickaxe.showConditioned(new Conditions(mi41, at40)
+		).isNotConsumed();
+	}
+
+	public void setupConditions()
+	{
+		inCopperZone = new ZoneRequirement(copperZone);
+		inIronZone = new ZoneRequirement(ironZone);
+	}
+
+	public void setupZones()
+	{
+		copperZone = new Zone(COPPER_POINT);
+		ironZone = new Zone(IRON_POINT);
+	}
+
+	private void setupSteps()
+	{
+		copperStep = new TileStep(this, COPPER_POINT,
+			"Stand next to the Copper- and Tin ore",
+			ironPickaxe, steelPickaxe, blackPickaxe);
+		copperStep.addTileMarker(COPPER_POINT, SpriteID.SKILL_MINING);
+
+		ironStep = new TileStep(this, IRON_POINT,
+			"Stand between the Iron ores",
+			steelPickaxe, blackPickaxe, mithrilPickaxe, adamantPickaxe, runePickaxe);
+		ironStep.addTileMarker(IRON_POINT, SpriteID.SKILL_MINING);
+
+
+		mineCopperOrTin = new ObjectStep(this, ObjectID.ROCKS_11360, COPPER_POINT,
+			"Mine Copper- and Tin ore at South-east Varrock mine until 15 Mining. You can choose to drop " +
+				"the ores as you go or bank them in the eastern Varrock bank.", true,
+			ironPickaxe, steelPickaxe, blackPickaxe
+		);
+		mineCopperOrTin.addAlternateObjects(ObjectID.ROCKS_11360, ObjectID.ROCKS_11161, ObjectID.ROCKS_10943);
```

Comment:
Could we amalgamate this step with the 'copperStep' step, and add the Tile Icon to this step? Would mean people can always see where to stand and helps avoid needing to check where the player is in this helper.

Sames goes for iron and the members guide.

Reply:
I'm not exactly sure what you want me to do 🙈

Reply:
So, rather than having 2 steps, 1 being 'Stand next to copper' and then another of 'mine copper', could we have these steps combined into 1 step of 'Go to X location and mine copper and tin', with a tile marker added to the step to indicate where to stand?